src/clj_sqs_extended/internal/receive.clj

```diff
@@ -66,31 +65,22 @@
 
 (defn- error-might-be-recovered-by-restarting
   [error]
-  (cond
-    (instance? ReflectiveOperationException error) false
-    (instance? RuntimeException error) false
-    (instance? UnknownHostException error) true
-    (instance? SocketException error) true
-    (instance? UnknownHostException error) true
-    (instance? HttpTimeoutException error) true
-    :else false))
+  (contains? #{UnknownHostException
+               SocketException
+               HttpTimeoutException} (type error)))
 
 (defn- handle-message-receival-error
   [sqs-ext-client loop-state error restart-delay-seconds receive-opts]
+  (log/warnf error "Error occured while receiving message!")
   (if (error-might-be-recovered-by-restarting error)
-    (let [count (get-in @loop-state [:stats :restart-count])]
-      (if (< count (:restart-limit receive-opts))
-        (do
-          (log/warnf error "Error occured while receiving message!")
-          (Thread/sleep (* 1000 restart-delay-seconds))
-          (restart-receive-loop sqs-ext-client
-                                loop-state
-                                receive-opts))
-        (log/warnf "Skipping receive-loop restart because the limit (%d) has been reached!"
-                   count)))
-    (do
-      (log/fatalf error "Unrecovereable error occured while receiving message!")
-      (stop-receive-loop loop-state))))
+    (when (< (get-in @loop-state [:stats :restart-count])
+             (:restart-limit receive-opts))
```

Comment:
what happens when restart-limit is reached? seems like this `when` should be an `if` like you had before, and print a warning that restart limit has reached.

stepping back, this nested `if` looks like it can be rewritten as a cond,

1) error is recoverable AND restart-limit not reached --> restart
2) error is recoverable BUT restart-limit is reached --> raise error `receive-loop for queue %s has reached its restart-limit [%d] with failure %s` (see my `ex-info` suggestion below)
3) errir is NOT recoverable --> raise error

Reply:
Yes, I made this change because I noticed that the log could get spammed with the 'reached limit message'. Your suggestion looks good, let me work that in.